### PR TITLE
Prefer the host dmd to compile the test runner

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
This ensures that all tests are run properly and potential failures (e.g. because of faulty codegen) occur in small snippets instead of the test runner itself.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: test/Makefile [hunk 1]
```diff
@@ -172,13 +206,13 @@ $(RESULTS_DIR)/d_do_test$(EXE): tools/d_do_test.d tools/sanitize_json.d $(RESULT
 	@echo "OS: '$(OS)'"
 	@echo "MODEL: '$(MODEL)'"
 	@echo "PIC: '$(PIC_FLAG)'"
-	$(DMD) -conf= $(MODEL_FLAG) $(PIC_FLAG) -g -lowmem -i -Itools -version=NoMain -unittest -run $<
-	$(DMD) -conf= $(MODEL_FLAG) $(PIC_FLAG) -g -lowmem -i -Itools -version=NoMain -od$(RESULTS_DIR) -of$@ $<
+	$(D) $(MODEL_FLAG) $(PIC_FLAG) -g -lowmem -i -Itools -version=NoMain -unittest -run $<
+	$(D) $(MODEL_FLAG) $(PIC_FLAG) -g -lowmem -i -Itools -version=NoMain -od$(RESULTS_DIR) -of$@ $<
```
Comment (Geod24) [inline test/Makefile]: Maybe use a slightly longer name here ?

Comment (Geod24) [reply test/Makefile]: Or not, since it was merged just now.

Comment (MoonlightSentinel) [reply test/Makefile]: Sure, wiill fix this in another PR.

Comment (MoonlightSentinel) [reply test/Makefile]: See #10977

**Pull request merged**